engine/executor/server.go

```diff
@@ -336,6 +350,20 @@ func (s *Server) Run(ctx context.Context) error {
 		return s.taskRunner.Run(ctx)
 	})
 
+	wg.Go(func() error {
```

Comment:
Can this be moved into the implementation of `jobAPIServer`? This way it would be easier for unit testing.

Reply:
Yes, it is more reasonable.